code/game/mecha/mecha.dm

```diff
@@ -146,7 +146,7 @@
 	internal_tank = new /obj/machinery/portable_atmospherics/canister/air(src)
 	return internal_tank
 
-/obj/mecha/proc/add_cell(var/obj/item/stock_parts/cell/C=null)
+/obj/mecha/proc/add_cell(var/obj/item/stock_parts/cell/high/plus/C=null)

```

Comment:
please fix the argument. This is really gross. 

Also, this isn't the proper way too do it, since this proc is utilized for more than just creating the cell.

You'd need to do `cell = new /obj/item/stock_parts/cell/high/plus(src)` and axe the `cell.charge` and `cell.maxcharge` things.